source/extensions/stat_sinks/common/statsd/statsd.h

```diff
@@ -78,6 +87,7 @@ class UdpStatsdSink : public Stats::Sink {
   const bool use_tag_;
   // Prefix for all flushed stats.
   const std::string prefix_;
+  const uint64_t buffer_size_;
```

Comment:
When you switch to the well known type I would recommend using absl::optional here to denote no default buffer size.

Reply:
I switched to the well-known type, and updated the constructor to make use of `absl::optional`, but I left this as a uint64_t and instead defaulted to a zero-buffer size at initialization. I personally think this keeps the actual buffer-writing code simpler because a write will always exceed the buffer-size, in which we simply perform a "best-effort" write. 

I'm not against using `absl::optional` throughout the class implementation, I just personally found this easier to reason about and figured I'd start here and get your thoughts.

Reply:
looks good